helm support: add github test lane

flavio-fernandes opened the pull request:
hi @jxiaobin : please rebase to latest master branch and update your pr to upstream ovnk repo

flavio-fernandes committed:
helm: Add GitHub Test Lane

This commit adds a new lane to test the GitHub
workflow, exercising the creation of an OVN
Kubernetes cluster via Helm. It introduces a
script called contrib/kind-helm.sh, which can
be used instead of contrib/kind.sh.

Even though the test lane has a placeholder
in the matrix for ovn-interconnect, this is not
being implemented in this commit.

Signed-off-by: Flavio Fernandes <ffernandes@nvidia.com>

jxiaobin commented on helm/ovn-kubernetes/charts/ovnkube-db/templates/deployment.yaml:
```diff
@@ -171,9 +171,29 @@ spec:
+                  - key: kubernetes.io/os
+                    operator: In
+                    values:
+                      - "linux"
```
How about we put the whole affinity in values.yaml, so that users can customize the spec as they wish?
i.e. in values.yaml we will have
```
affinity:
  nodeAffinity:
    requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                  - key: node-role.kubernetes.io/control-plane
                    operator: Exists
                  - key: kubernetes.io/os
                    operator: In
                    values:
                      - "linux"```
in deployment.yaml:
```
{{- with .Values.affinity }}
affinity:
{{- toYaml . | nindent 8 }} // 8 is an example
{{- end }}
```
example: https://github.com/influxdata/helm-charts/blob/master/charts/telegraf/templates/deployment.yaml#L111

flavio-fernandes replied:
fixed

jxiaobin commented on helm/ovn-kubernetes/charts/ovnkube-db/templates/deployment.yaml:
```diff
@@ -171,9 +171,29 @@ spec:
+                    operator: In
+                    values:
+                      - "linux"
+        podAntiAffinity:
```
same as above

jxiaobin commented on helm/ovn-kubernetes/charts/ovnkube-identity/templates/ovnkube-identity.yaml:
```diff
@@ -36,15 +36,16 @@ spec:
-      # required to be scheduled on a linux node with node-role.kubernetes.io/control-plane label and
-      # only one instance of ovnkube-control-plane pod per node
+      # required to be scheduled on a linux node and only one instance of ovnkube-identity pod per node
       affinity:
```
same as above

flavio-fernandes replied:
fixed

jxiaobin commented on helm/ovn-kubernetes/charts/ovnkube-master/templates/deployment-ovnkube-master.yaml:
```diff
@@ -40,17 +40,16 @@ spec:
-      # required to be scheduled on a linux node with node-role.kubernetes.io/control-plane label and
-      # only one instance of ovnkube-master pod per node
+      # required to be scheduled on a linux node and only one instance of ovnkube-master pod per node
       affinity:
```
same as above

flavio-fernandes replied:
fixed

jxiaobin commented on helm/ovn-kubernetes/charts/ovnkube-zone-controller/templates/ovnkube-zone-controller.yaml:
```diff
@@ -35,20 +35,29 @@ spec:
       {{- end }}
-      # required to be scheduled on a linux node with node-role.kubernetes.io/zone-controller label
+      # required to be scheduled on a linux node and only one instance of ovnkube-zone-controller pod per node
       affinity:
```
same as above

flavio-fernandes replied:
fixed

flavio-fernandes committed:
helm: Move pod affinity to values.yaml

Allow greater flexibility in which affinity to use, so that pods
in the control plane can be provisioned via custom values.

Signed-off-by: Flavio Fernandes <ffernandes@nvidia.com>

jxiaobin commented:
LGTM
